contracts/Groth16Verifier.sol

```diff
@@ -20,107 +20,66 @@
 pragma solidity 0.8.23;
 import "./Groth16.sol";
 
-library Pairing {
-  // The prime q in the base field F_q for G1
-  uint private constant _Q =
+contract Groth16Verifier is IGroth16Verifier {
+  uint private constant _P =
     21888242871839275222246405745257275088696311157297823662689037894645226208583;
+  uint256 private constant _Q =
```

Comment:
This one should definitely be called `R` instead of `Q`. Using `r` for the size of the scalar field (the one ending with ...17) is essentially an universal standard. For the other one (here `P`) some people use `p` and some people use `q`).